bool    towerFound
int32    towerPos
int32    towerDistance
int32    towerBlobHeight
int32    towerBlobWidth
bool    factoryFound
int32    factoryPos
int32    factoryDistance
int32    factoryBlobHeight
int32    factoryBlobWidth
